.github/workflows/tests.yml

```diff
@@ -1,46 +1,42 @@
-name: Test Suite
+name: Build
+
 on:
   push:
     branches:
       - master
   pull_request:
 jobs:
-    test:
-        runs-on: ubuntu-latest
-        strategy:
-            matrix:
-                php: [ 7.1, 7.2, 7.3, 7.4, '8.0' ]
-        name: "PHP ${{ matrix.php }} Unit Test"
-        steps:
-            - uses: actions/checkout@v2
-            - uses: codecov/codecov-action@v1
-            - name: Setup PHP
-              uses: shivammathur/setup-php@v2
-              with:
-                php-version: ${{ matrix.php }}
-            - name: Install composer dependencies
-              uses: nick-invision/retry@v1
-              with:
-                timeout_minutes: 10
-                max_attempts: 3
-                command: composer install
-            - name: Run script
-              run: vendor/bin/phpunit
-
-    style:
-        runs-on: ubuntu-latest
-        name: PHP Style Check
-        steps:
-            - uses: actions/checkout@v2
-            - name: Setup PHP
-              uses: shivammathur/setup-php@v2
-              with:
-                php-version: 7.4
-            - name: Install Dependencies
-              uses: nick-invision/retry@v1
-              with:
-                timeout_minutes: 10
-                max_attempts: 3
-                command: composer install
-            - name: Run Script
-              run: vendor/bin/php-cs-fixer fix --dry-run --diff .
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions:
+          - '7.4'
+          - '8.0'
+          - '8.1'
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+      -
+        name: Install dependencies
+        run: composer install ${{ matrix.composer-options }}
+      -
+        name: Tests
+        run: composer ci
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Build docker image
+        run: |
+          docker-compose build tests
```

Comment:
tady to k něčemu potřebuješ? Přijde mě že to moc nedává smysl tady v té lib. tests nepoužíváš pro testy a image nikam neposíláš. Ten tests container bych odstranil úplně z docker-compose a nechal bych dev. Tady bych pustil ten build dev že sa ti ten container zbuildí pro dev.